### PR TITLE
Component generator - Functional Components and Pure Components

### LINKED ISSUE #3: Option for generating functional/stateless components
When running `ignite g component ExampleComponent`, there should be an option to make the component "dumb", i.e. make the component a function rather than a class.

### PR DESCRIPTION
**Pull request by evang9410**
Added options in the component generator to generate a `PureComponent` or a`FunctionalComponent`

Fixes #3

### CODE REVIEW

File: templates/component-functional.ejs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { StyleSheet } from 'react-native';
+
+import Text from '<%= props.appName %>/app/components/base/Text';
+import View from '<%= props.appName %>/app/components/base/View';
+
+const styles = StyleSheet.create({
+    container: {},
+});
+
+type Props = {};
```
Comment (robertcoopercode) [inline templates/component-functional.ejs]: New line after `type Props = {};` plz.

File: templates/component.ejs [hunk 2]
```diff
@@ -14,7 +14,8 @@ const styles = StyleSheet.create({
 
 type Props = {};
 
-<% if (props.isConnected) { %>class <%= props.name %> extends Component<Props> {<% } else { %>export default class <%= props.name %> extends Component<Props> {<% } %>
+<% if (props.isConnected) { %><% if (props.isPure) { %>class <%= props.name %> extends PureComponent<Props> {<% } else { %>class <% props.name %> extends Component<Props> {<% } %><% } else { %><% if (props.isPure) { %>export default class <%= props.name %> extends PureComponent<Props> {<% } else { %>export default class <% props.name %> extends Component<Props> {<% } %><% } %>
```
Comment (adrienthiery) [inline templates/component.ejs]: You could define a variable in ejs in here : 
`<% var typeOfComponent = props.isPure ? 'PureComponent' : 'Component' %>`
and use that everywhere

**Review by adrienthiery — CHANGES_REQUESTED** [review]

File: templates/component.ejs (hunk 2 above)
Comment (evang9410) [reply templates/component.ejs]: @adrienthiery At last, I got around to it, thanks!

**Review by nathanielram — APPROVED** [review]

File: templates/component.ejs [hunk 3]
```diff
@@ -13,15 +13,16 @@ const styles = StyleSheet.create({
 });
 
 type Props = {};
+<% var typeOfComponent = props.isPure ? 'PureComponent' : 'Component' %>
```
Comment (adrienthiery) [inline templates/component.ejs]: This should go all the way on top, that way you can clean the `react` import :)

File: templates/component.ejs [hunk 4]
```diff
@@ -13,15 +13,16 @@ const styles = StyleSheet.create({
...
+<% if (props.isConnected) { %>class <%= props.name %> extends <%= typeOfComponent %><Props> {<% } else { %>export default class <%= props.name %> extends <%= typeOfComponent %><Props> {<% } %>
```
Comment (adrienthiery) [inline templates/component.ejs]: Same thing could be applied for `export default class` vs `class` if `props.isConnected`

**Review by adrienthiery — CHANGES_REQUESTED** [review]

File: templates/component.ejs [hunk 5]
```diff
@@ -11,11 +11,10 @@ import View from '<%= props.appName %>/app/components/base/View';
 const styles = StyleSheet.create({
     container: {},
 });
-
-type Props = {};
 <% var typeOfComponent = props.isPure ? 'PureComponent' : 'Component' %>
-<% if (props.isConnected) { %>class <%= props.name %> extends <%= typeOfComponent %><Props> {<% } else { %>export default class <%= props.name %> extends <%= typeOfComponent %><Props> {<% } %>
-
+type Props = {};
+<% var classDeclaration = props.isConnected ? 'class ' + props.name + ' extends ' + typeOfComponent + '<Props> {' : 'export default class ' + props.name + ' extends ' + typeOfComponent + '<Props> {'%>
```
Comment (adrienthiery) [inline templates/component.ejs]: 🤔 That looks more complicated/less readable than what it should be.

What do you think of : 
```
type Props = {};
<% var classDeclaration = props.isConnected ? 'class' : 'export default class' %>
<%= classDeclaration %> <%= props.name %> extends <%= typeOfComponent %><Props> {
    // Defaults for props
    static defaultProps = {};
```

Comment (evang9410) [reply templates/component.ejs]: Much more readable. I updated.

File: templates/component.ejs [hunk 6]
```diff
@@ -11,17 +11,17 @@ import View from '<%= props.appName %>/app/components/base/View';
 const styles = StyleSheet.create({
     container: {},
 });
-
+<% var typeOfComponent = props.isPure ? 'PureComponent' : 'Component' %>
```
Comment (adrienthiery) [inline templates/component.ejs]: If you're defining that all the way on top after `// @flow`, you can use `typeOfComponent` in the `import React, { <%= typeOfComponent %> } from 'react';` 😛

**Review by adrienthiery — CHANGES_REQUESTED** [review]
🕷

**Comment by evang9410** [comment]
🐜

**Review by adrienthiery — APPROVED** [review]

**Pull request merged**